src/textual/widget.py

```diff
@@ -982,11 +982,13 @@ def post_render(self, renderable: RenderableType) -> RenderableType:
         """
 
         if isinstance(renderable, str):
-            renderable = Text.from_markup(renderable)
+            renderable = Text.from_markup(renderable, justify=self.styles.text_justify)
 
         rich_style = self.rich_style
         if isinstance(renderable, Text):
             renderable.stylize(rich_style)
+            if not renderable.justify:
```

Comment:
Wasn't sure whether to do this `if` or not...

Reply:
Seems reasonable...